### PR TITLE
Fix tag parsing in DCA tags injection 

### PR DESCRIPTION
**Pull request by sarah-witt**
### What does this PR do?

Fixes tag parsing when tags have spaces in them when performing DCA tags injection

Additionally, cleans up some of the logic regarding logs config creation, and adds more log lines.

### Review checklist (to be filled by reviewers)

- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: lib/scripts/get_tags.py [hunk 1]
```diff
@@ -56,14 +59,14 @@
 user_tags = os.environ.get('DD_TAGS', None)
 if user_tags:
     try:
+        user_tags = user_tags.replace(" ", "_")
         user_tags = user_tags.split(',')
         for tag in user_tags:
             tags.append(tag)
     except Exception as e:
         print("there was an issue parsing the tags in DD_TAGS: {}".format(e))
```
Comment (NouemanKHAL) [inline lib/scripts/get_tags.py:68]:
```suggestion
tags = [ tag.replace("", " _") for tag in tags ]
```
Let's do it once here instead.
I didn't test the following code but it should be close.

File: lib/scripts/get_tags.py [hunk 2]
```diff
@@ -27,13 +27,15 @@
     if node_agent_tags is not None:
         # we do this to separate commas inside json values from tags separator commas
         node_agent_tags = node_agent_tags.replace(",\"", ";\"")
+        node_agent_tags = node_agent_tags.replace(" ", "_")
```
Comment (NouemanKHAL) [inline lib/scripts/get_tags.py:30]: Let's remove these

File: lib/scripts/get_tags.py [hunk 3]
```diff
@@ -27,13 +27,15 @@
...
         all_node_agent_tags = node_agent_tags.split(",")
         tags = tags + [tag for tag in all_node_agent_tags if ";" not in tag]
 
 
 for vcap_var_name in vcap_variables:
     vcap_var = vcap_application.get(vcap_var_name)
     if vcap_var is not None:
+        vcap_var = vcap_var.replace(" ", "_")
```
Comment (NouemanKHAL) [inline lib/scripts/get_tags.py:38]: And this

File: lib/scripts/get_tags.py [hunk 4]
```diff
@@ -48,6 +50,7 @@
 if user_tags:
     try:
         user_tags = user_tags.split(',')
+        user_tags = user_tags.replace(" ", "_")
```
Comment (NouemanKHAL) [inline lib/scripts/get_tags.py:53]: This too

File: lib/scripts/get_tags.py [hunk 5]
```diff
@@ -56,14 +59,15 @@
 user_tags = os.environ.get('DD_TAGS', None)
 if user_tags:
     try:
+        user_tags = user_tags.replace(" ", "_")
```
Comment (NouemanKHAL) [inline lib/scripts/get_tags.py:62]: And finally this

File: lib/scripts/parse_env_vars.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import sys
+
+if len(sys.argv) > 1:
+    env_file_name = sys.argv[1]
+    with open(env_file_name, 'r') as env_file:
+        env_vars = env_file.readlines()
+
+    if len(sys.argv) > 2:
+        new_env_file_name = sys.argv[2]
+        with open(new_env_file_name, 'w') as new_env_file:
```
Comment (NouemanKHAL) [inline lib/scripts/parse_env_vars.py:14]:
```suggestion
        with open(new_env_file_name, 'w+') as new_env_file:
```
How about we create the file if it doesn't exist and remove the `touch` from the bash script?

File: lib/scripts/update_agent_config_restart.sh [hunk 7]
```diff
@@ -13,15 +13,18 @@ DATADOG_DIR="${DATADOG_DIR:-/home/vcap/app/.datadog}"
 SUPPRESS_DD_AGENT_OUTPUT="${SUPPRESS_DD_AGENT_OUTPUT:-true}"
 
 # correct way to export / source the .datadog_env file so that every variable is parsed
-export $(grep -v '^#' $DATADOG_DIR/.datadog_env | xargs)
+touch "$DATADOG_DIR/.new_datadog_env"
```
Comment (NouemanKHAL) [inline lib/scripts/update_agent_config_restart.sh:16]:
```suggestion
```

We won't need this if we use `w+` mode  in the python script.

File: lib/scripts/parse_env_vars.py [hunk 8]
```diff
@@ -11,12 +11,11 @@
 
     if len(sys.argv) > 2:
         new_env_file_name = sys.argv[2]
-        with open(new_env_file_name, 'w') as new_env_file:
+        with open(new_env_file_name, 'w+') as new_env_file:
             for env_var in env_vars:
                 env_var = env_var.strip()
                 if env_var.startswith("#"):
                     continue
-                env_var = env_var.replace(" ", "_")
```
Comment (NouemanKHAL) [inline lib/scripts/parse_env_vars.py]: Why did you remove this?

Comment (sarah-witt) [reply lib/scripts/parse_env_vars.py]: I accidentally removed this and meant to add this back when I was testing, sorry about that!

**Review by NouemanKHAL — APPROVED** [review]
